### DISCUSSION TITLE
How to decompress .basis.navmesh?

### DISCUSSION THREAD

**Discussion by Rendok** [facebookresearch/habitat-lab#1489] [Q&A]
Hi,

I have multiple *.basis.navmesh from HM3D. I want to try writing different pathfinding algorithms for educational purposes. So how can I decompress the navmesh in terminal or python?

**Comment by aclegg3** [comment]
Hey @Rendok,

In python, you can open the navmesh with habitat-sim [Pathfinder](https://aihabitat.org/docs/habitat-sim/habitat_sim.nav.PathFinder.html#load_nav_mesh).
Then you can extract the mesh [vertices](https://aihabitat.org/docs/habitat-sim/habitat_sim.nav.PathFinder.html#build_navmesh_vertices) and [indices](https://aihabitat.org/docs/habitat-sim/habitat_sim.nav.PathFinder.html#build_navmesh_vertex_indices) and save them to whatever format you like.

```
my_pathfinder = PathFinder()
my_pathfinder.load_nav_mesh(navmesh_filepath)
verts = my_pathfinder.build_navmesh_vertices()
indices = my_pathfinder.build_navmesh_vertex_indices()
```
